lms/djangoapps/course_home_api/progress/v1/serializers.py

```diff
@@ -1,9 +1,62 @@
 """
 Progress Tab Serializers
 """
-
+import uuid
 from rest_framework import serializers
 from lms.djangoapps.course_home_api.outline.v1.serializers import CourseBlockSerializer
+from rest_framework.reverse import reverse
+
+
+class CoursewareSummarySerializer(serializers.Serializer):
+    """
+    Serializer
+    """
+    blocks = serializers.SerializerMethodField()
+
+    def get_blocks(self, blocks):
+        return [
+            {
+                'display_name': chapter['display_name'],
+                'chapter_url': reverse(  # currently points to the LMS url not the MFE
+                    'jump_to',
+                    kwargs={'course_id': str(self.context['course_key']), 'location': str(chapter['url_name'])},
+                    request=self.context['request'],
+                ),
+                'sections': [
+                    {
+                        'display_name': str(section.display_name),
+                        'section_url': reverse(
+                            'jump_to',
+                            kwargs={'course_id': str(self.context['course_key']), 'location': str(section.url_name)},
+                            request=self.context['request'],
+                        ),
+                        'percent_graded': section.percent_graded,
+                        'graded_total': {
+                            'first_attempted': section.graded_total.first_attempted,
+                            'possible': section.graded_total.possible,
+                            'graded': section.graded_total.graded,
+                            'earned': section.graded_total.earned,
+                        },
```

Comment:
I think this can just be `'graded_total': section.graded_total`

Reply:
graded_total is actually an object which isn't JSON serializable!